pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -1,726 +1,690 @@
-  ${optionalString (versionOlder version "4.11") ''
-    VIDEO_STK1160_AC97 y
-  ''}
-
-  # Our initrd init uses shebang scripts, so can't be modular.
-  BINFMT_SCRIPT y
-
-  # For systemd-binfmt
-  BINFMT_MISC? y
-
-  # Enable the 9P cache to speed up NixOS VM tests.
-  9P_FSCACHE? y
-  9P_FS_POSIX_ACL? y
-
-  # Enable transparent support for huge pages.
-  TRANSPARENT_HUGEPAGE? y
-  TRANSPARENT_HUGEPAGE_ALWAYS? n
-  TRANSPARENT_HUGEPAGE_MADVISE? y
-
-  # zram support (e.g for in-memory compressed swap).
-  ZRAM m
-  ZSWAP? y
-  ZBUD? y
-  ${optionalString (versionOlder version "3.18") ''
-    ZSMALLOC y
-  ''}
-  ${optionalString (versionAtLeast version "3.18") ''
-    ZSMALLOC m
-  ''}
-
-  # Enable PCIe and USB for the brcmfmac driver
-  BRCMFMAC_USB? y
-  BRCMFMAC_PCIE? y
-
-  # Support x2APIC (which requires IRQ remapping).
-  ${optionalString (stdenv.system == "x86_64-linux") ''
-    X86_X2APIC y
-    IRQ_REMAP y
-  ''}
-
-  # needed for iwd WPS support (wpa_supplicant replacement)
-  ${optionalString (versionAtLeast version "4.7") ''
-    KEY_DH_OPERATIONS y
-  ''}
-
-  # Disable the firmware helper fallback, udev doesn't implement it any more
-  FW_LOADER_USER_HELPER_FALLBACK? n
-
-  # Disable various self-test modules that have no use in a production system
-  # This menu disables all/most of them on >= 4.16
-  RUNTIME_TESTING_MENU? n
-  # For older kernels, painstakingly disable each symbol.
-  ${optionalString (versionOlder version "4.16") ''
-    ARM_KPROBES_TEST? n
-    ASYNC_RAID6_TEST? n
-    ATOMIC64_SELFTEST? n
-    BACKTRACE_SELF_TEST? n
-    INTERVAL_TREE_TEST? n
-    PERCPU_TEST? n
-    RBTREE_TEST? n
-    TEST_BITMAP? n
-    TEST_BPF? n
-    TEST_FIRMWARE? n
-    TEST_HASH? n
-    TEST_HEXDUMP? n
-    TEST_KMOD? n
-    TEST_KSTRTOX? n
-    TEST_LIST_SORT? n
-    TEST_LKM? n
-    TEST_PARMAN? n
-    TEST_PRINTF? n
-    TEST_RHASHTABLE? n
-    TEST_SORT? n
-    TEST_STATIC_KEYS? n
-    TEST_STRING_HELPERS? n
-    TEST_UDELAY? n
-    TEST_USER_COPY? n
-    TEST_UUID? n
-  ''}
-
-  CRC32_SELFTEST? n
-  CRYPTO_TEST? n
-  ${optionalString (versionOlder version "4.18") ''
-    DRM_DEBUG_MM_SELFTEST? n
-    LNET_SELFTEST? n
-  ''}
-  EFI_TEST? n
-  GLOB_SELFTEST? n
-  LOCK_TORTURE_TEST? n
-  MTD_TESTS? n
-  NOTIFIER_ERROR_INJECTION? n
-  RCU_PERF_TEST? n
-  RCU_TORTURE_TEST? n
-  TEST_ASYNC_DRIVER_PROBE? n
-  WW_MUTEX_SELFTEST? n
-  XZ_DEC_TEST? n
-
-  ${optionalString (features.criu or false)  ''
-    EXPERT y
-    CHECKPOINT_RESTORE y
-  ''}
-
-  ${optionalString ((features.criu or false) && (features.criu_revert_expert or true))
-    # Revert some changes, introduced by EXPERT, when necessary for criu
-  ''
-    RFKILL_INPUT? y
-    HID_PICOLCD_FB? y
-    HID_PICOLCD_BACKLIGHT? y
-    HID_PICOLCD_LCD? y
-    HID_PICOLCD_LEDS? y
-    HID_PICOLCD_CIR? y
-    DEBUG_MEMORY_INIT? y
-  ''}
-
-  ${extraConfig}
-''
+with import ../../../../lib/kernel.nix { inherit (stdenv) lib; inherit version; };
+
+let
+  # temporary hack
```

Comment:
Hmm, what's this for?

Reply:
when reworking the original PR, the features system had changed a bit and I came up with the hack to remove later. I had forgotten about it; I pushed a 3rd commit which uses features.xdom0 instead. It should fix it.